src/components/TransactionSettingsBottomSheet.tsx

```diff
@@ -0,0 +1,183 @@
+import { useTransactionSettingsStore } from "ducks/transactionSettings";
+import useAppTranslation from "hooks/useAppTranslation";
+import useColors from "hooks/useColors";
+import { useNetworkFees } from "hooks/useNetworkFees";
+import { useValidateMemo } from "hooks/useValidateMemo";
+import React, { useState } from "react";
+import { TouchableOpacity, View } from "react-native";
+
+type TransactionSettingsBottomSheetProps = {
+  onCancel: () => void;
+  onConfirm: () => void;
+};
+
+const TransactionSettingsBottomSheet: React.FC<
+  TransactionSettingsBottomSheetProps
+> = ({ onCancel, onConfirm }) => {
+  const { t } = useAppTranslation();
+  const {
+    transactionMemo,
+    saveMemo,
+    transactionTimeout,
+    saveTransactionTimeout,
+    transactionFee,
+    saveTransactionFee,
+  } = useTransactionSettingsStore();
+  const { themeColors } = useColors();
+  const { recommendedFee, networkCongestion } = useNetworkFees();
+  const [localFee, setLocalFee] = useState(transactionFee ?? recommendedFee);
+  const [localMemo, setLocalMemo] = useState(transactionMemo);
+  const [localTimeout, setLocalTimeout] = useState(
+    transactionTimeout.toString(),
+  );
+  const { error } = useValidateMemo(localMemo);
+
+  const handleSetLocalFee = (value: string) => {
+    const fee = Math.max(Number(value), Number(MIN_TRANSACTION_FEE));
+    setLocalFee(fee.toString());
+  };
+
+  const handleConfirm = () => {
+    if (error) return;
+    saveMemo(localMemo);
+    saveTransactionTimeout(Number(localTimeout));
+    saveTransactionFee(localFee);
+    onConfirm();
+  };
+
+  const getLocalizedCongestionLevel = (
+    congestion: NetworkCongestion,
+  ): string => {
+    switch (congestion) {
+      case NetworkCongestion.LOW:
+        return t("low");
+      case NetworkCongestion.MEDIUM:
+        return t("medium");
+      case NetworkCongestion.HIGH:
+        return t("high");
+      default:
+        return t("low");
+    }
+  };
+
+  return (
+    <View className="flex-1">
+      <View className="flex-1 justify-between">
+        <View className="flex flex-col gap-2">
+          <View className="flex flex-row items-center gap-2">
+            <Text sm secondary>
+              {t("transactionSettingsBottomSheet.feeTitle")}
+            </Text>
+            <TouchableOpacity onPress={() => {}}>
+              {/* disabled until info is added */}
+              {/* <Icon.InfoCircle color={themeColors.gray[8]} size={16} />  */}
+            </TouchableOpacity>
+          </View>
+          <View className="flex flex-row mt-[4px] items-center gap-2">
+            <Input
+              isBottomSheetInput
+              fieldSize="lg"
+              value={localFee}
+              leftElement={
+                <Icon.Route size={16} color={themeColors.foreground.primary} />
+              }
+              onChangeText={handleSetLocalFee}
+              keyboardType="numeric"
+              placeholder={MIN_TRANSACTION_FEE}
+              error={error}
```

Comment:
it looks like the Fee and Memo fields are sharing the same error message, is this expected?